// @pulumi/pulumiservice/deploymentSchedule.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeploymentSchedule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * A scheduled recurring or single time run of a pulumi command.
 */
class DeploymentSchedule extends pulumi.CustomResource {
    /**
     * Get an existing DeploymentSchedule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DeploymentSchedule(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DeploymentSchedule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DeploymentSchedule.__pulumiType;
    }
    /**
     * Create a DeploymentSchedule resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.organization === undefined) && !opts.urn) {
                throw new Error("Missing required property 'organization'");
            }
            if ((!args || args.project === undefined) && !opts.urn) {
                throw new Error("Missing required property 'project'");
            }
            if ((!args || args.pulumiOperation === undefined) && !opts.urn) {
                throw new Error("Missing required property 'pulumiOperation'");
            }
            if ((!args || args.stack === undefined) && !opts.urn) {
                throw new Error("Missing required property 'stack'");
            }
            resourceInputs["organization"] = args ? args.organization : undefined;
            resourceInputs["project"] = args ? args.project : undefined;
            resourceInputs["pulumiOperation"] = args ? args.pulumiOperation : undefined;
            resourceInputs["scheduleCron"] = args ? args.scheduleCron : undefined;
            resourceInputs["stack"] = args ? args.stack : undefined;
            resourceInputs["timestamp"] = args ? args.timestamp : undefined;
            resourceInputs["scheduleId"] = undefined /*out*/;
        }
        else {
            resourceInputs["organization"] = undefined /*out*/;
            resourceInputs["project"] = undefined /*out*/;
            resourceInputs["pulumiOperation"] = undefined /*out*/;
            resourceInputs["scheduleCron"] = undefined /*out*/;
            resourceInputs["scheduleId"] = undefined /*out*/;
            resourceInputs["stack"] = undefined /*out*/;
            resourceInputs["timestamp"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(DeploymentSchedule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DeploymentSchedule = DeploymentSchedule;
/** @internal */
DeploymentSchedule.__pulumiType = 'pulumiservice:index:DeploymentSchedule';
