// @pulumi/pulumiservice/driftSchedule.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DriftSchedule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * A cron schedule to run drift detection.
 */
class DriftSchedule extends pulumi.CustomResource {
    /**
     * Get an existing DriftSchedule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new DriftSchedule(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DriftSchedule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DriftSchedule.__pulumiType;
    }
    /**
     * Create a DriftSchedule resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a;
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.organization === undefined) && !opts.urn) {
                throw new Error("Missing required property 'organization'");
            }
            if ((!args || args.project === undefined) && !opts.urn) {
                throw new Error("Missing required property 'project'");
            }
            if ((!args || args.scheduleCron === undefined) && !opts.urn) {
                throw new Error("Missing required property 'scheduleCron'");
            }
            if ((!args || args.stack === undefined) && !opts.urn) {
                throw new Error("Missing required property 'stack'");
            }
            resourceInputs["autoRemediate"] = (_a = (args ? args.autoRemediate : undefined)) !== null && _a !== void 0 ? _a : false;
            resourceInputs["organization"] = args ? args.organization : undefined;
            resourceInputs["project"] = args ? args.project : undefined;
            resourceInputs["scheduleCron"] = args ? args.scheduleCron : undefined;
            resourceInputs["stack"] = args ? args.stack : undefined;
            resourceInputs["scheduleId"] = undefined /*out*/;
        }
        else {
            resourceInputs["autoRemediate"] = undefined /*out*/;
            resourceInputs["organization"] = undefined /*out*/;
            resourceInputs["project"] = undefined /*out*/;
            resourceInputs["scheduleCron"] = undefined /*out*/;
            resourceInputs["scheduleId"] = undefined /*out*/;
            resourceInputs["stack"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(DriftSchedule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DriftSchedule = DriftSchedule;
/** @internal */
DriftSchedule.__pulumiType = 'pulumiservice:index:DriftSchedule';
