// @pulumi/pulumiservice/teamEnvironmentPermission.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TeamEnvironmentPermission = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * A permission for a team to use an environment.
 */
class TeamEnvironmentPermission extends pulumi.CustomResource {
    /**
     * Get an existing TeamEnvironmentPermission resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new TeamEnvironmentPermission(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of TeamEnvironmentPermission.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TeamEnvironmentPermission.__pulumiType;
    }
    /**
     * Create a TeamEnvironmentPermission resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.environment === undefined) && !opts.urn) {
                throw new Error("Missing required property 'environment'");
            }
            if ((!args || args.organization === undefined) && !opts.urn) {
                throw new Error("Missing required property 'organization'");
            }
            if ((!args || args.permission === undefined) && !opts.urn) {
                throw new Error("Missing required property 'permission'");
            }
            if ((!args || args.team === undefined) && !opts.urn) {
                throw new Error("Missing required property 'team'");
            }
            resourceInputs["environment"] = args ? args.environment : undefined;
            resourceInputs["organization"] = args ? args.organization : undefined;
            resourceInputs["permission"] = args ? args.permission : undefined;
            resourceInputs["team"] = args ? args.team : undefined;
        }
        else {
            resourceInputs["permission"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(TeamEnvironmentPermission.__pulumiType, name, resourceInputs, opts);
    }
}
exports.TeamEnvironmentPermission = TeamEnvironmentPermission;
/** @internal */
TeamEnvironmentPermission.__pulumiType = 'pulumiservice:index:TeamEnvironmentPermission';
